common/txmgr/types/fee_estimator.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import "context"
+
+// Opt is an option for a gas estimator
+type Opt int
+
+const (
+	// OptForceRefetch forces the estimator to bust a cache if necessary
+	OptForceRefetch Opt = iota
+)
+
+// PriorAttempt provides a generic interface for reading tx data to be used in the fee esimators
+type PriorAttempt[FEE any, HASH any] interface {
+	Fee() FEE
+	GetChainSpecificGasLimit() uint32
+	GetBroadcastBeforeBlockNum() *int64
+	GetHash() HASH
+	GetTxType() int
+}
+
+// FeeEstimator provides a generic interface for fee estimation
+//
+//go:generate mockery --quiet --name FeeEstimator --output ./mocks/ --case=underscore
+type FeeEstimator[HEAD any, FEE any, MAXPRICE any, HASH any] interface {
```

Comment:
The MAXPRICE actually is also never needed to be passed in to estimator by the generic txmgr.
The Txmgr just reads a static config, and passes it in. Ideally it should be fetched directly by the estimator.

I would say, create a sub-task or ticket, to remove it from generic interface here later.

Reply:
How would it be fetched directly?

Reply:
This is used on a job-specific basis

Reply:
I would agree with Sam, I think it's still needed as it's a different config variable then the global chain configs and can be different for each tx

Reply:
but then each gas estimator would need knowledge of the `FromAddress` for each key. it seems simpler to just pass gas parameters instead of having the estimators have knowledge of additional configs + addresses